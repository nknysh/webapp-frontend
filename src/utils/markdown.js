import MarkdownIt from 'markdown-it';

const md = new MarkdownIt({ html: true });

export const parseMarkdown = markdown => md.parse(markdown);
export const renderMarkdown = markdown => md.render(markdown);
