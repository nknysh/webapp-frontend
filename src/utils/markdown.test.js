import { parseMarkdown, renderMarkdown } from './markdown';

describe('markdown', () => {
  it('parseMarkdown returns parsed markdown', () => {
    expect(parseMarkdown('### Hello')).toMatchSnapshot();
  });
  it('renderMarkdown returns rendererd markdown', () => {
    expect(renderMarkdown('### Hello')).toMatchSnapshot();
  });
});
