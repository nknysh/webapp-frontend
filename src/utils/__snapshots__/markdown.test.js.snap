// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown parseMarkdown returns parsed markdown 1`] = `
Array [
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": Array [
      0,
      1,
    ],
    "markup": "###",
    "meta": null,
    "nesting": 1,
    "tag": "h3",
    "type": "heading_open",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": Array [
      Token {
        "attrs": null,
        "block": false,
        "children": null,
        "content": "Hello",
        "hidden": false,
        "info": "",
        "level": 0,
        "map": null,
        "markup": "",
        "meta": null,
        "nesting": 0,
        "tag": "",
        "type": "text",
      },
    ],
    "content": "Hello",
    "hidden": false,
    "info": "",
    "level": 1,
    "map": Array [
      0,
      1,
    ],
    "markup": "",
    "meta": null,
    "nesting": 0,
    "tag": "",
    "type": "inline",
  },
  Token {
    "attrs": null,
    "block": true,
    "children": null,
    "content": "",
    "hidden": false,
    "info": "",
    "level": 0,
    "map": null,
    "markup": "###",
    "meta": null,
    "nesting": -1,
    "tag": "h3",
    "type": "heading_close",
  },
]
`;

exports[`markdown renderMarkdown returns rendererd markdown 1`] = `
"<h3>Hello</h3>
"
`;
